mantis-server/mantis-server-worker/src/main/java/io/mantisrx/server/worker/jobmaster/WorkerMetricHandler.java

```diff
@@ -280,6 +307,28 @@ public void call() {
                                                     gauges.get(ON_NEXT_GAUGE) / 6.0, numWorkers, ""));
                                 }
                             }
+
+                            double sourceJobDrops = 0;
+                            boolean hasSourceJobDropsMetric = false;
+                            Map<String, String> sourceMetricsRecent = sourceJobMetricsRecent.asMap();
+                            for (Map.Entry<String, WorkerMetrics> worker : sourceJobWorkersMap.entrySet()) {
+                                Map<String, GaugeData> metricGroups = metricAggregator.getAggregates(worker.getValue().getGaugesByMetricGrp());
+                                for (Map.Entry<String, GaugeData> group : metricGroups.entrySet()) {
+                                    String metricKey = worker.getKey() + ":" + group.getKey();
+                                    Map<String, Double> gauges = group.getValue().getGauges();
+                                    if (sourceMetricsRecent.containsKey(metricKey) && gauges.containsKey(DROPPED_COUNTER_METRIC_NAME)) {
```

Comment:
gauges list won't have the Dropped counter, so this condition will be false always. There is a separate counters list in the metric event. Though, the step based autoscaler has only looked at gauges so far, so the aggregations of the metric data for a single worker and across workers might have some assumptions based on that.

Reply:
The metric gets converted to gauges here: https://github.com/Netflix/mantis/pull/82/files/e489031772bf021ee83fe0c145e1e202dbc0638a#diff-fce5244027e72de847d3bffc7dfdd658a28fd5ab90b802b0dd662268a1112f2bR100

This is to simplify implementation and similar to how the stage ON_NEXT_GAUGE metric is generated. The other option is to make `MetricData`, `WorkerMetric` classes counter aware. That would be a bigger change.

Reply:
oh ok, I missed the counter to gauge conversion, so was confused reading the code.